docs/dev_guide.md

```diff
@@ -1,31 +1,149 @@
 # Dev Guide
 
-## Adding a new param in GCSFuse
+## Adding new features to GCSFuse
+
+**If you're interested in adding a new feature, please open a GitHub issue to
+discuss your proposal with the GCSFuse team first before making the
+contribution.**
+
+Rollout of any new feature should be done in 2 phases:
+
+**1. Experimental Phase:**
+
+* New features should initially be introduced behind an experimental flag. These
+  flags should be prefixed with `--experimental-` and marked as hidden in the
+  help output.
+* This signals to users that the feature is in development and may change or be
+  removed.
+* After the feature development is complete, developers should open a GitHub
+  issue (
+  see [example](https://github.com/GoogleCloudPlatform/gcsfuse/issues/793)) to
+  track the feature's progress towards production readiness. This issue should
+  outline the motivation for the feature, its design, and any known limitations.
+
+**2. Production Rollout:**
```

Comment:
So will the old PR be merged and this is in a follow-up PR.

Reply:
Yes, do you think this needs to be explicitly mentioned?

Reply:
Yea feel its easy to get confused what is production rollout for us vs them.
For the customers production rollout could mean its just available in a production version or next version.
For us, i guess the meaning here is for all users availability. ( this could be outside the next immediate version ).